sklearn/ensemble/_hist_gradient_boosting/tests/test_loss.py

```diff
@@ -211,11 +218,33 @@ def test_baseline_categorical_crossentropy():
         assert np.allclose(baseline_prediction[k, :], np.log(p))
 
 
+def test_baseline_poisson_loss():
+    rng = np.random.RandomState(0)
+
+    loss = _LOSSES['poisson_loss'](sample_weight=None)
+    y_train = rng.poisson(size=100).astype(np.float64)
+    # Make sure at least one sample point is larger than zero
+    y_train[0] = 1.
+    baseline_prediction = loss.get_baseline_prediction(y_train, None, 1)
+    assert baseline_prediction.shape == tuple()  # scalar
+    assert baseline_prediction.dtype == y_train.dtype
+    assert_all_finite(baseline_prediction)
+    # Make sure baseline prediction produces the mean of all targets
+    y_baseline = loss.inverse_link_function(baseline_prediction)
+    assert_almost_equal(np.mean(y_baseline), y_train.mean())
```

Comment:
For consistency with the other checks we should directly compare `baseline_prediction`

```suggestion
    # Make sure baseline prediction produces the log of the mean of all targets
    assert_almost_equal(np.log(y_train.mean()), baseline_prediction)
```